### ISSUE TITLE
After Upgrade to 4.14 - all Cluster members not reachable

### ISSUE THREAD

**Issue by ninjuan11** [lxc/lxd#8834]
After updating vom 4.06 to 4.14 my cluster completly broke.
only message in lxc cluster list is "No heartbeat for xxx" on all members, even itself.
Maybe because of new cluster certs? Didnt find a migration document. Need help how to get my cluster responsive again.

```
config:
  cluster.https_address: 172.16.0.193:9443
  core.https_address: 172.16.0.193:9443
  core.trust_password: true
  images.auto_update_interval: "15"
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- macaroon_authentication
- network_sriov
- console
- restrict_devlxd
- migration_pre_copy
- infiniband
- maas_network
- devlxd_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
- devlxd_images
- container_local_cross_pool_handling
- proxy_unix
- proxy_udp
- clustering_join
- proxy_tcp_udp_multi_port_handling
- network_state
- proxy_unix_dac_properties
- container_protection_delete
- unix_priv_drop
- pprof_http
- proxy_haproxy_protocol
- network_hwaddr
- proxy_nat
- network_nat_order
- container_full
- candid_authentication
- backup_compression
- candid_config
- nvidia_runtime_config
- storage_api_volume_snapshots
- storage_unmapped
- projects
- candid_config_key
- network_vxlan_ttl
- container_incremental_copy
- usb_optional_vendorid
- snapshot_scheduling
- snapshot_schedule_aliases
- container_copy_project
- clustering_server_address
- clustering_image_replication
- container_protection_shift
- snapshot_expiry
- container_backup_override_pool
- snapshot_expiry_creation
- network_leases_location
- resources_cpu_socket
- resources_gpu
- resources_numa
- kernel_features
- id_map_current
- event_location
- storage_api_remote_volume_snapshots
- network_nat_address
- container_nic_routes
- rbac
- cluster_internal_copy
- seccomp_notify
- lxc_features
- container_nic_ipvlan
- network_vlan_sriov
- storage_cephfs
- container_nic_ipfilter
- resources_v2
- container_exec_user_group_cwd
- container_syscall_intercept
- container_disk_shift
- storage_shifted
- resources_infiniband
- daemon_storage
- instances
- image_types
- resources_disk_sata
- clustering_roles
- images_expiry
- resources_network_firmware
- backup_compression_algorithm
- ceph_data_pool_name
- container_syscall_intercept_mount
- compression_squashfs
- container_raw_mount
- container_nic_routed
- container_syscall_intercept_mount_fuse
- container_disk_ceph
- virtual-machines
- image_profiles
- clustering_architecture
- resources_disk_id
- storage_lvm_stripes
- vm_boot_priority
- unix_hotplug_devices
- api_filtering
- instance_nic_network
- clustering_sizing
- firewall_driver
- projects_limits
- container_syscall_intercept_hugetlbfs
- limits_hugepages
- container_nic_routed_gateway
- projects_restrictions
- custom_volume_snapshot_expiry
- volume_snapshot_scheduling
- trust_ca_certificates
- snapshot_disk_usage
- clustering_edit_roles
- container_nic_routed_host_address
- container_nic_ipvlan_gateway
- resources_usb_pci
- resources_cpu_threads_numa
- resources_cpu_core_die
- api_os
- container_nic_routed_host_table
- container_nic_ipvlan_host_table
- container_nic_ipvlan_mode
- resources_system
- images_push_relay
- network_dns_search
- container_nic_routed_limits
- instance_nic_bridged_vlan
- network_state_bond_bridge
- usedby_consistency
- custom_block_volumes
- clustering_failure_domains
- resources_gpu_mdev
- console_vga_type
- projects_limits_disk
- network_type_macvlan
- network_type_sriov
- container_syscall_intercept_bpf_devices
- network_type_ovn
- projects_networks
- projects_networks_restricted_uplinks
- custom_volume_backup
- backup_override_name
- storage_rsync_compression
- network_type_physical
- network_ovn_external_subnets
- network_ovn_nat
- network_ovn_external_routes_remove
- tpm_device_type
- storage_zfs_clone_copy_rebase
- gpu_mdev
- resources_pci_iommu
- resources_network_usb
- resources_disk_address
- network_physical_ovn_ingress_mode
- network_ovn_dhcp
- network_physical_routes_anycast
- projects_limits_instances
- network_state_vlan
- instance_nic_bridged_port_isolation
- instance_bulk_state_change
- network_gvrp
- instance_pool_move
- gpu_sriov
- pci_device_type
- storage_volume_state
- network_acl
- migration_stateful
- disk_state_quota
- storage_ceph_features
- projects_compression
- projects_images_remote_cache_expiry
- certificate_project
- network_ovn_acl
- projects_images_auto_update
- projects_restricted_cluster_target
- images_default_architecture
- network_ovn_acl_defaults
- gpu_mig
- project_usage
- network_bridge_acl
- warnings
- projects_restricted_backups_and_snapshots
- clustering_join_token
- clustering_description
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
environment:
  addresses:
  - 172.16.0.193:9443
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
XXXXXXXXXXX
    -----END CERTIFICATE-----
  certificate_fingerprint: xxx
  driver: lxc | qemu
  driver_version: 4.0.9 | 5.2.0
  firewall: nftables
  kernel: Linux
  kernel_architecture: x86_64
  kernel_features:
    netnsid_getifaddrs: "true"
    seccomp_listener: "true"
    seccomp_listener_continue: "true"
    shiftfs: "false"
    uevent_injection: "true"
    unpriv_fscaps: "true"
  kernel_version: 5.10.0-0.bpo.3-amd64
  lxc_features:
    cgroup2: "true"
    devpts_fd: "true"
    mount_injection_file: "true"
    network_gateway_device_route: "true"
    network_ipvlan: "true"
    network_l2proxy: "true"
    network_phys_macvlan_mtu: "true"
    network_veth_router: "true"
    pidfd: "true"
    seccomp_allow_deny_syntax: "true"
    seccomp_notify: "true"
    seccomp_proxy_send_notify_fd: "true"
  os_name: Debian GNU/Linux
  os_version: "10"
  project: default
  server: lxd
  server_clustered: true
  server_name: cluster-00
  server_pid: 165814
  server_version: "4.14"
  storage: ceph | dir
  storage_version: ceph version 12.2.13 (584a20eb0237c657dc0567da126be145106aa47e)
    luminous (stable) | 1
```

**Comment by tomponline** [comment]
Please can you enable debug mode and see what is being logged in the LXD log?

```
sudo snap set lxd daemon.debug=true; sudo systemctl reload snap.lxd.daemon
sudo tail -f /var/snap/lxd/common/lxd/logs/lxd.log
```

**Comment by tomponline** [comment]
I just tried upgrading a LXD 4.0.6 (pre-per-server-certificates) cluster to a LXD 4.14 cluster and it worked fine.

**Comment by ninjuan11** [comment]
```
t=2021-06-01T18:39:25+0200 lvl=info msg="LXD 4.14 is starting in normal mode" path=/var/snap/lxd/common/lxd
t=2021-06-01T18:39:25+0200 lvl=info msg="Kernel uid/gid map:" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - u 0 0 4294967295" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - g 0 0 4294967295" 
t=2021-06-01T18:39:25+0200 lvl=info msg="Configured LXD uid/gid map:" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - u 0 1000000 1000000000" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - g 0 1000000 1000000000" 
t=2021-06-01T18:39:25+0200 lvl=info msg="Kernel features:" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - closing multiple file descriptors efficiently: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - netnsid-based network retrieval: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - pidfds: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - uevent injection: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - seccomp listener: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - seccomp listener continue syscalls: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - seccomp listener add file descriptors: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - attach to namespaces via pidfds: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - safe native terminal allocation : yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - unprivileged file capabilities: yes" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - cgroup layout: hybrid" 
t=2021-06-01T18:39:25+0200 lvl=warn msg=" - Couldn't find the CGroup blkio.weight, disk priority will be ignored" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - shiftfs support: disabled" 
t=2021-06-01T18:39:25+0200 lvl=info msg="Initializing local database" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Refreshing local trusted certificate cache" 
t=2021-06-01T18:39:25+0200 lvl=info msg="Set client certificate to server certificate e19623e335ba49179d1840f9d7bd1dcaabfe116fad0b5ab9c3cbb2d9bdcf8bc9" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Initializing database gateway" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Start database node" address=172.16.0.193:9443 id=1 role=spare
t=2021-06-01T18:39:25+0200 lvl=info msg="Starting cluster handler:" 
t=2021-06-01T18:39:25+0200 lvl=info msg="Starting /dev/lxd handler:" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - binding devlxd socket" socket=/var/snap/lxd/common/lxd/devlxd/sock
t=2021-06-01T18:39:25+0200 lvl=info msg="REST API daemon:" 
t=2021-06-01T18:39:25+0200 lvl=info msg=" - binding Unix socket" inherited=true socket=/var/snap/lxd/common/lxd/unix.socket
t=2021-06-01T18:39:25+0200 lvl=info msg=" - binding TCP socket" socket=172.16.0.193:9443
t=2021-06-01T18:39:25+0200 lvl=info msg="Initializing global database" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Matched trusted cert" fingerprint=e19623e335ba49179d1840f9d7bd1dcaabfe116fad0b5ab9c3cbb2d9bdcf8bc9 subject="CN=root@cluster-00,O=linuxcontainers.org"
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.193:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.193:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.194:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.194:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.195:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.195:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.196:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.196:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.197:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.197:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.198:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.198:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.199:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.199:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.200:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.200:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.201:9443
t=2021-06-01T18:39:25+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.201:9443: no known leader" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.202:9443
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Dqlite: attempt 0: server 172.16.0.202:9443: connect to reported leader 172.16.0.203:9443" 
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Triggering an out of schedule hearbeat" address=172.16.0.203:9443
t=2021-06-01T18:39:25+0200 lvl=dbug msg="Dqlite: attempt 0: server 172.16.0.202:9443: connected" 
t=2021-06-01T18:39:25+0200 lvl=info msg="Firewall loaded driver \"nftables\"" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.194:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.196:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.195:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.200:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.199:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.197:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.198:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.203:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.202:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.201:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.204:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.205:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.207:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.208:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Notify node 172.16.0.206:9443 of state changes" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote LXD over HTTPs" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Initializing storage pools" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initializing and checking storage pool \"default\"" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Mount started" driver=dir pool=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Mount finished" driver=dir pool=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initializing and checking storage pool \"ceph\"" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Mount started" driver=ceph pool=ceph
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Mount finished" driver=ceph pool=ceph
t=2021-06-01T18:39:26+0200 lvl=info msg="Initializing daemon storage mounts" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Initializing networks" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg=Start driver=bridge network=lxdbr0 project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Setting up network" driver=bridge network=lxdbr0 project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Stable MAC generated" driver=bridge hwAddr=00:16:3e:08:bf:77 network=lxdbr0 project=default seed=e19623e335ba49179d1840f9d7bd1dcaabfe116fad0b5ab9c3cbb2d9bdcf8bc9.0.1
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Clearing firewall" driver=bridge network=lxdbr0 project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Setting up firewall" driver=bridge network=lxdbr0 project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="New task Operation: 68bae430-5f94-4edf-8fb8-797d1dbac63f" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Pruning leftover image files" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Started task operation: 68bae430-5f94-4edf-8fb8-797d1dbac63f" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Done pruning leftover image files" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Loading daemon configuration" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Success for task operation: 68bae430-5f94-4edf-8fb8-797d1dbac63f" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initialized inotify with file descriptor 21" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initialising mounted disk ref counter" device=root driver=disk instance=monitor project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="MountInstance started" driver=ceph instance=monitor pool=ceph project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="MountInstance finished" driver=ceph instance=monitor pool=ceph project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initialising mounted disk ref counter" device=root driver=disk instance=vpn project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="MountInstance started" driver=ceph instance=vpn pool=ceph project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="MountInstance finished" driver=ceph instance=vpn pool=ceph project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initialising mounted disk ref counter" device=root driver=disk instance=webc00-lb project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="MountInstance started" driver=ceph instance=webc00-lb pool=ceph project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="MountInstance finished" driver=ceph instance=webc00-lb pool=ceph project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initialising mounted disk ref counter" device=shared_storage driver=disk instance=webc00-lb project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initialising mounted disk ref counter" device=root driver=disk instance=webc01-lb project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="MountInstance started" driver=ceph instance=webc01-lb pool=ceph project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="MountInstance finished" driver=ceph instance=webc01-lb pool=ceph project=default
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Initialising mounted disk ref counter" device=shared_storage driver=disk instance=webc01-lb project=default
t=2021-06-01T18:39:26+0200 lvl=info msg="Started seccomp handler" path=/var/snap/lxd/common/lxd/seccomp.socket
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Refreshing trusted certificate cache" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg=Handling ip=@ method=GET protocol=unix url=/1.0 username=root
t=2021-06-01T18:39:26+0200 lvl=dbug msg="New task Operation: 16b886ca-2c39-4b67-8534-0da1afe5d52a" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Pruning expired images" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Started task operation: 16b886ca-2c39-4b67-8534-0da1afe5d52a" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Done pruning expired images" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Success for task operation: 16b886ca-2c39-4b67-8534-0da1afe5d52a" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="New task Operation: df94334f-b340-40fa-a461-df8d2ac36a01" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Pruning expired instance backups" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Started task operation: df94334f-b340-40fa-a461-df8d2ac36a01" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Done pruning expired instance backups" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Success for task operation: df94334f-b340-40fa-a461-df8d2ac36a01" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="New task Operation: 48fc40c2-d90a-4b8f-aace-fd1160d3f7ca" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="New task Operation: f6a6bb37-1b23-44e9-8d67-bebd4b135a57" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Pruning resolved warnings" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Started task operation: f6a6bb37-1b23-44e9-8d67-bebd4b135a57" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="New task Operation: da5a8494-221b-46c4-a113-e8b56e231d58" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Skipping image synchronization task since we're not leader" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="New task Operation: cc7cd39b-0215-4a75-9c14-111ec26dae27" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Expiring log files" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Started task operation: da5a8494-221b-46c4-a113-e8b56e231d58" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Success for task operation: f6a6bb37-1b23-44e9-8d67-bebd4b135a57" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg=Handling ip=@ method=GET protocol=unix url=/internal/ready username=root
t=2021-06-01T18:39:26+0200 lvl=info msg="Updating instance types" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Started task operation: 48fc40c2-d90a-4b8f-aace-fd1160d3f7ca" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Done expiring log files" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Done updating instance types" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Updating images" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Started task operation: cc7cd39b-0215-4a75-9c14-111ec26dae27" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Done pruning resolved warnings" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Processing image" alias=debian/10 fingerprint=49ccee803693d1898dd1decf091e689bf887016316423d346008fe5a7f9bb00d protocol=simplestreams server=https://images.linuxcontainers.org
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Connecting to a remote simplestreams server" 
t=2021-06-01T18:39:26+0200 lvl=info msg="Done updating images" 
t=2021-06-01T18:39:26+0200 lvl=dbug msg="Success for task operation: da5a8494-221b-46c4-a113-e8b56e231d58" 
t=2021-06-01T18:39:27+0200 lvl=dbug msg="Acquiring lock for image download of \"49ccee803693d1898dd1decf091e689bf887016316423d346008fe5a7f9bb00d\"" 
t=2021-06-01T18:39:27+0200 lvl=dbug msg="Lock acquired for image download of \"49ccee803693d1898dd1decf091e689bf887016316423d346008fe5a7f9bb00d\"" 
t=2021-06-01T18:39:27+0200 lvl=eror msg="Failed to update the image" err="Failed locating image \"49ccee803693d1898dd1decf091e689bf887016316423d346008fe5a7f9bb00d\" in the cluster: Image not available on any online node" fingerprint=49ccee803693d1898dd1decf091e689bf887016316423d346008fe5a7f9bb00d
t=2021-06-01T18:39:27+0200 lvl=dbug msg="Success for task operation: cc7cd39b-0215-4a75-9c14-111ec26dae27" 
t=2021-06-01T18:39:27+0200 lvl=dbug msg=Handling ip=@ method=GET protocol=unix url=/1.0 username=root
t=2021-06-01T18:39:27+0200 lvl=dbug msg=Handling ip=@ method=GET protocol=unix url=/internal/ready username=root
t=2021-06-01T18:39:28+0200 lvl=dbug msg="Success for task operation: 48fc40c2-d90a-4b8f-aace-fd1160d3f7ca" 
t=2021-06-01T18:39:56+0200 lvl=dbug msg=Handling ip=@ method=GET protocol=unix url=/1.0 username=nagios
t=2021-06-01T18:39:56+0200 lvl=dbug msg=Handling ip=@ method=GET protocol=unix url=/1.0/cluster username=nagios
t=2021-06-01T18:39:56+0200 lvl=dbug msg=Handling ip=@ method=GET protocol=unix url="/1.0/cluster/members?recursion=1" username=nagios
```

```
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
|   NAME   |            URL            | DATABASE | ARCHITECTURE | FAILURE DOMAIN | DESCRIPTION |  STATE  |                                   MESSAGE                                    |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-00 | https://172.16.0.193:9443 | NO       | x86_64       | default        | master      | OFFLINE | No heartbeat for 45m12.527066262s (2021-06-01 17:56:43.723998555 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-01 | https://172.16.0.194:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.539581647s (2021-06-01 17:56:43.715328926 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-02 | https://172.16.0.195:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.547349112s (2021-06-01 17:56:43.713720051 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-03 | https://172.16.0.196:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.539511094s (2021-06-01 17:56:43.718599352 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-04 | https://172.16.0.197:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.555776444s (2021-06-01 17:56:43.714338752 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-05 | https://172.16.0.198:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.544860531s (2021-06-01 17:56:43.728898528 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-06 | https://172.16.0.199:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.54925345s (2021-06-01 17:56:43.717585919 +0200 CEST)  |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-07 | https://172.16.0.200:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.548838017s (2021-06-01 17:56:43.715003601 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-08 | https://172.16.0.201:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.57922149s (2021-06-01 17:56:43.703314653 +0200 CEST)  |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-09 | https://172.16.0.202:9443 | YES      | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.573754598s (2021-06-01 17:56:43.705906306 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-10 | https://172.16.0.203:9443 | YES      | x86_64       | default        |             | OFFLINE | No heartbeat for 45m42.551174001s (2021-06-01 17:56:13.725542873 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-11 | https://172.16.0.204:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.571239116s (2021-06-01 17:56:43.713975724 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-12 | https://172.16.0.205:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.578177361s (2021-06-01 17:56:43.709885186 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-13 | https://172.16.0.206:9443 | YES      | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.575895633s (2021-06-01 17:56:43.720660963 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-14 | https://172.16.0.207:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.569188462s (2021-06-01 17:56:43.722227038 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
| cluster-15 | https://172.16.0.208:9443 | NO       | x86_64       | default        |             | OFFLINE | No heartbeat for 45m12.578721321s (2021-06-01 17:56:43.715697972 +0200 CEST) |
+----------+---------------------------+----------+--------------+----------------+-------------+---------+------------------------------------------------------------------------------+
```

**Comment by ninjuan11** [comment]
```
+--------------+---------------+------------------------------+------------------------------+
| FINGERPRINT  |  COMMON NAME  |          ISSUE DATE          |         EXPIRY DATE          |
+--------------+---------------+------------------------------+------------------------------+
| 7a0789d8f4ae | root@cluster-05 | Sep 3, 2020 at 9:22pm (UTC)  | Sep 1, 2030 at 9:22pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| 8f9b0f0e64f6 | root@cluster-07 | Sep 3, 2020 at 9:22pm (UTC)  | Sep 1, 2030 at 9:22pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| 54bc4aaf430f | root@cluster-03 | Sep 3, 2020 at 9:22pm (UTC)  | Sep 1, 2030 at 9:22pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| 63a8ef389a94 | root@cluster-13 | Oct 12, 2020 at 4:22pm (UTC) | Oct 10, 2030 at 4:22pm (UTC) |
+--------------+---------------+------------------------------+------------------------------+
| 93a417b54cc5 | root@cluster-08 | Sep 3, 2020 at 9:26pm (UTC)  | Sep 1, 2030 at 9:26pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| 99a74abb5605 | root@cluster-06 | Sep 3, 2020 at 9:22pm (UTC)  | Sep 1, 2030 at 9:22pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| 233dcb570a54 | root@cluster-10 | Sep 3, 2020 at 9:22pm (UTC)  | Sep 1, 2030 at 9:22pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| 927a98fcc52e | root@cluster-09 | Sep 3, 2020 at 9:26pm (UTC)  | Sep 1, 2030 at 9:26pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| bbb0ae40fb4d | root@cluster-01 | Sep 3, 2020 at 9:20pm (UTC)  | Sep 1, 2030 at 9:20pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| d6fbc0eaa5d8 | root@cluster-12 | Oct 12, 2020 at 4:22pm (UTC) | Oct 10, 2030 at 4:22pm (UTC) |
+--------------+---------------+------------------------------+------------------------------+
| e2ff1f231680 | root@cluster-11 | Sep 11, 2020 at 9:49pm (UTC) | Sep 9, 2030 at 9:49pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| e98fc9e8a62c | root@cluster-02 | Sep 3, 2020 at 9:22pm (UTC)  | Sep 1, 2030 at 9:22pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| e19623e335ba | root@cluster-00 | Sep 3, 2020 at 8:07pm (UTC)  | Sep 1, 2030 at 8:07pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
| f096a7cbb6a3 | root@cluster-15 | Oct 12, 2020 at 4:22pm (UTC) | Oct 10, 2030 at 4:22pm (UTC) |
+--------------+---------------+------------------------------+------------------------------+
| fad95ad8c988 | root@cluster-14 | Oct 12, 2020 at 4:22pm (UTC) | Oct 10, 2030 at 4:22pm (UTC) |
+--------------+---------------+------------------------------+------------------------------+
| fc960c95e4ea | root@cluster-16 | Jun 1, 2021 at 3:53pm (UTC)  | May 30, 2031 at 3:53pm (UTC) |
+--------------+---------------+------------------------------+------------------------------+
| fcecccb555d2 | root@cluster-04 | Sep 3, 2020 at 9:22pm (UTC)  | Sep 1, 2030 at 9:22pm (UTC)  |
+--------------+---------------+------------------------------+------------------------------+
```

**Comment by tomponline** [comment]
Thanks. There does appear to be an issue indeed. The cluster certificate upgrade completes successfully, but the heartbeat process doesn't seem to restart again (unlike when upgrading from 4.13 to 4.14). Perhaps there is some difference between the way the heartbeats are initiated in the LTS branch.

**Comment by ninjuan11** [comment]
any idea how to fix this? as this is a productive cluster, this is pretty bad for me - i had to update because i added two new servers and snap removed the older images - i was only able to pick 4.13 or 4.14.

**Comment by tomponline** [comment]
@ninjuan11 for future reference you can pick the LTS branch when installing LXD using `sudo snap install lxd --channel=4.0/stable`. See https://discuss.linuxcontainers.org/t/managing-the-lxd-snap/8178 for more info around that.

As for fixing it, let me see if reloading LXD fixes...one moment please.

**Comment by tomponline** [comment]
@ninjuan11 also can you still run `lxc ls` btw?

**Comment by ninjuan11** [comment]
yes lxc ls works, but only local vms are visible, rest is error.

**Comment by tomponline** [comment]
@ninjuan11 ok good so the dqlite cluster (and by extension the certificate trust store) is working OK, so it appears to be just the lack of heartbeats that is the issue.

**Comment by tomponline** [comment]
@ninjuan11 I just tried the upgrade from 4.0.6 to 4.14 again and it succeeded this time, suggesting the issue may be if the time between upgrading the members was far apart the cluster may have become out of sync. Can you have a look at the logs on your other members and see when they got upgrade to 4.14 please, and how long apart that happened?

**Comment by ninjuan11** [comment]
one member took a little longer to upgrade - about 10 min - but all members are on 4.14 right now.

**Comment by tomponline** [comment]
@ninjuan11 do you see any `database is locked` in the logs?

**Comment by ninjuan11** [comment]
i dont see anything regarding locked database in the logfiles.

i remember seeing in a logfile an error like "unable to inform server that database has been upgraded" (nopt exact wording) but i cant find it anymroe

**Comment by tomponline** [comment]
@ninjuan11 in my experiments I found that I could do `sudo systemctl reload snap.lxd.daemon` on all the nodes except one of them which hung in the logs with an error about locking. I think if you try and reload all of the members and then `kill -9` the `lxd` process of any that won't reload cleanly then that should fix it (it has on my recreation), as getting lxd to restart allows the heartbeat process to begin again.

As for the cause I'll need to look into this in more detail.

**Comment by ninjuan11** [comment]
i dumped all logfiles right now, seems like i misstyped the regular expression before - i have multiple ones with different locks

```
grep -e "database is locked" log*
log03:t=2021-06-01T19:20:35+0200 lvl=dbug msg="Database error: protocol.Error{Code:5, Message:\"database is locked\"}" 
log03:t=2021-06-01T19:20:35+0200 lvl=dbug msg="Retry failed db interaction (database is locked)" 
log04:t=2021-06-01T19:20:35+0200 lvl=dbug msg="Database error: protocol.Error{Code:5, Message:\"database is locked\"}" 
log04:t=2021-06-01T19:20:35+0200 lvl=dbug msg="Retry failed db interaction (database is locked)" 
log09:t=2021-06-01T19:20:35+0200 lvl=dbug msg="Database error: &errors.errorString{s:\"insert or replaced row: database is locked\"}" 
log09:t=2021-06-01T19:20:35+0200 lvl=dbug msg="Retry failed db interaction (insert or replaced row: database is locked)" 
log11:t=2021-06-01T19:20:34+0200 lvl=dbug msg="Database error: failed to update node version info: database is locked" 
log11:t=2021-06-01T19:20:34+0200 lvl=dbug msg="Retry failed db interaction (failed to update node version info: database is locked)" 
```

i reloaded all except machine 11 - didnt help :-(

**Comment by tomponline** [comment]
@ninjuan11 and did you kill -9 lxd process and restart lxd on that machine 11?

**Comment by tomponline** [comment]
@ninjuan11 basically you need to restart LXD, so you need to kill and restart any of the ones that wont reload due to be locked.

**Comment by ninjuan11** [comment]
ah misread the instruction, thought i should kill only these who dont reload correctly. i killed it and issued a systemctl restart snap.lxd.daemon - but it hangs.

**Comment by tomponline** [comment]
@ninjuan11 it looks like you have a few that are locked so need to kill and restart  it on each of them.

**Comment by ninjuan11** [comment]
i did it on every machine, all are hanging on the restart command.

**Comment by ninjuan11** [comment]
it seems like now i have another bug to report. because i restarted the machines, new leaders got elected on the last 3 in the cluster. the restarted machines gave up after checking the first 10.

**Comment by ninjuan11** [comment]
after i tried to restart one of the leaders:
```
t=2021-06-01T18:56:47+0200 lvl=info msg="Done pruning expired instance backups"
t=2021-06-01T18:56:47+0200 lvl=info msg="Done updating images"
t=2021-06-01T19:20:33+0200 lvl=info msg="Received 'terminated signal', waiting for all operations to finish"
t=2021-06-01T19:20:34+0200 lvl=info msg="All running operations finished, shutting down"
t=2021-06-01T19:20:38+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: call exec-sql (budget 0s): receive: header: EOF"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.193:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.194:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.195:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.196:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.197:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.198:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.199:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.200:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.201:9443: no known leader"
t=2021-06-01T19:20:43+0200 lvl=warn msg="Dqlite: attempt 0: server 172.16.0.202:9443: no known leader"
t=2021-06-01T19:32:21+0200 lvl=warn msg="Could not handover member's responsibilities: Post \"https://172.16.0.196:9443/internal/cluster/assign\": EOF"
t=2021-06-01T19:32:21+0200 lvl=info msg="Starting shutdown sequence"
t=2021-06-01T19:32:23+0200 lvl=warn msg="Give up waiting to get number of running containers"
t=2021-06-01T19:32:23+0200 lvl=info msg="Closing the database"
t=2021-06-01T19:32:23+0200 lvl=warn msg="Failed to get number of running containers: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=info msg="Stop database gateway"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
t=2021-06-01T19:32:23+0200 lvl=eror msg="Error updating" err="Failed to check if this node is out-of-date: failed to begin transaction: sql: database is closed"
```

this killed everything on all machines...

**Comment by tomponline** [comment]
@ninjuan11 but have you killed that lxd process using `kill -9 <process id>` - its clear the DB is locked and you need to forceably kill the process so it can restart. Doing `systemctl` won't work.

**Comment by ninjuan11** [comment]
yes, i killed them with kill -9.

**Comment by tomponline** [comment]
@ninjuan11 can you try this, shutdown all of the lxd processes using `lxd shutdown` and then on one of the DB members run

```
lxd recover-from-quorum-loss
```

Then start that lxd member up and check it has started cleanly and that your instances are listed.

Then start each member one by one after that.

**Comment by ninjuan11** [comment]
i just sent the command (lxd shutdown) to all members of the cluster, takes very long to complete... i'll post here when i finished the task.

**Comment by tomponline** [comment]
@ninjuan11 ok @stgraber has just suggested that `recover-from-quorum-loss` isn't whats needed.

**Comment by ninjuan11** [comment]
i'm very concerned that i'm having a data loss right now. last backup is a few weeks old, would be terrible.

**Comment by tomponline** [comment]
@ninjuan11 so I think we need to be clear about this. Have you stopped entirely (i.e you can confirmed no lxd process running) on all nodes and then started a single one up and check it starts cleanly?

**Comment by ninjuan11** [comment]
its still in the task of stopping. lxd shutdown takes horribly long.

**Comment by tomponline** [comment]
@ninjuan11 you probably need to do a `kill -9` as it'll be hanging trying to hand over the role. You need to ensure they have all been entirely stopped before starting one up.

**Comment by ninjuan11** [comment]
i got zombies.
 231336 ?        Z      0:00 [lxd] <defunct>

cant kill them anymore...

**Comment by tomponline** [comment]
@ninjuan11 ps aux of your host please

**Comment by ninjuan11** [comment]
because it is so long: 
https://nopaste.net/EsXIqb1frI

**Comment by tomponline** [comment]
@ninjuan11 looks like you've been killing the wrong process:

Kill 179373

**Comment by ninjuan11** [comment]
i already killed this multiple times, always instanenious respawn

**Comment by tomponline** [comment]
Try disabling the  lxd service with systemctl disable snap.lxd.daemon service then kill it

**Comment by ninjuan11** [comment]
already did, just took a little bit. everything is down now. i'll start the recover process

**Comment by ninjuan11** [comment]
uhm - 
```
root@cluster-15 /home/ # /snap/bin/lxd recover-from-quorum-loss
Error: unknown command "recover-from-quorum-loss" for "lxd"
```

**Comment by tomponline** [comment]
I asked you not to run that earlier as i sought advice. I'm not sure what to do now.

You needed to shutdown everything and restart because it doesn't look like you achieved that earlier

**Comment by ninjuan11** [comment]
sorry i was a little bit puzzled what to do now, but it seems like the command was not triggered because it tried to attach to the zombie.

**Comment by ninjuan11** [comment]
seems like snap startet lxd and ignored systemctl disable. the cluster found all nodes after a good old reboot, but `lxc ls` is not working anymore
```
lxc cluster list
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
|   NAME   |            URL            | DATABASE | ARCHITECTURE | FAILURE DOMAIN | DESCRIPTION | STATE  |      MESSAGE      |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-00 | https://172.16.0.193:9443 | NO       | x86_64       | default        | master      | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-01 | https://172.16.0.194:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-02 | https://172.16.0.195:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-03 | https://172.16.0.196:9443 | YES      | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-04 | https://172.16.0.197:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-05 | https://172.16.0.198:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-06 | https://172.16.0.199:9443 | YES      | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-07 | https://172.16.0.200:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-08 | https://172.16.0.201:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-09 | https://172.16.0.202:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-10 | https://172.16.0.203:9443 | YES      | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-11 | https://172.16.0.204:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-12 | https://172.16.0.205:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-13 | https://172.16.0.206:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-14 | https://172.16.0.207:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
| cluster-15 | https://172.16.0.208:9443 | NO       | x86_64       | default        |             | ONLINE | Fully operational |
+----------+---------------------------+----------+--------------+----------------+-------------+--------+-------------------+
```

**Comment by stgraber** [comment]
A few simple things to check without getting too deep into things:
 - Make sure each server is able to reach all the others (`curl -k -s https://172.16.0.X:8443` to every one of the other 14 IPs)
 - Make sure all servers have an accurate clock (they should all be within a 2s margin)

If that's all good, then you can try:
 - `lxc info --resources` to confirm local API is functional (doesn't use DB)
 - `lxc info --resources --target OTHER-SERVER` to check API access to each of the other servers

Then on each server, run `lxd sql global "SELECT COUNT(id) FROM nodes"` to check whether the global database is working.
If that still looks fine, then run `lxc list --fast`, if some are in ERROR, then run `lxc info NAME` on one of those.

**Comment by ninjuan11** [comment]
i restarted the whole cluster very very slowly.. it worked .. for a few minutes. everything was available with all machines in.
a few minutes later a lot of the virtual machines are on error - but only if i use `lxc list`. with `lxc list --fast` everything is fine.

**Comment by stgraber** [comment]
How many instances do you have and how many snapshots do they typically have?
CEPH clusters that haven't been tweaked can be extremely slow at getting snapshot size data (`rbd du`) which could have you hit some timeouts.

**Comment by ninjuan11** [comment]
i have about 30 vms on 16 hosts with 1-2 snapshots each. worked flawless before

**Comment by stgraber** [comment]
Yeah but you upgraded from `4.0.6` to `4.14` which added a lot of features, including the fact that snapshot sizes are now retrieved from ceph.

**Comment by stgraber** [comment]
If `lxc list --fast` works fine but `lxc list` doesn't and `lxc info` itself is slow when querying snapshots, the problem is almost certainly your ceph cluster being very slow at retrieving snapshot sizes.

You can likely confirm that by just looking for `ceph du` processes showing up on all your systems while running that `lxc list`.

You could try to get an idea of what's going on by running just `lxc query /1.0/instances/NAME/snapshots/SNAPNAME` (picking one of your instances and one of its snapshots). That should get you an idea of what it takes to get the size data of just one snapshot and if things fail, hopefully get you a clearer error.

**Comment by ninjuan11** [comment]
i didnt have any performance issues before with ceph, so this is really a drawback for me. i can indeed see the rbd du process when trying to do a lxc query on snapshots. how can i disable this feature? i really dont need to see the sizes of each snapshot everytime i do a list command.

**Comment by tomponline** [comment]
Glad to hear that the LXD cluster is back up and running after a fresh restart, I will look into why the upgrade doesn't always work properly. Although it sounds like there is a different ceph issue now unrelated to the original issue.

**Comment by ninjuan11** [comment]
it cant be the ceph. when issuing the rbd du command by hand it willl return the value in less than 1 second. still issuing the lxc query snapshot command will never return a value.

**Comment by tomponline** [comment]
@ninjuan11 it can be quite slow as it will issue one of those commands for each instance and each snapshot. Do you see any errors now in your debug log?

**Comment by ninjuan11** [comment]
it is not issuing this command on each snapshot right now, i issued `time lxc query /1.0/instances/con-00/snapshots/running --verbose --debug` - so i doubt it has something to do with ceph right now.

**Comment by tomponline** [comment]
@ninjuan11 but to confirm the output of `lxc cluster ls` - the original problem is resolved now right? The heartbeats are going again?

**Comment by ninjuan11** [comment]
yes, its working again, but the bug is still a problem as this should never happen in the first place :-/ 
the problem was indeed that one of the 16 servers did not upgrade in a timely manner (high load), which resulted in this behaviour. i would suggest the script should retry after a few minutes instead of giving up completly and lock up.

 i'll create a new ticket regarding the other problem.

**Comment by tomponline** [comment]
@ninjuan11 thanks, yes I agree, I am actively working on fixing the original issue. Thanks for opening another issue regarding the ceph performance problem.

**Comment by stgraber** [comment]
Can you check `ps fauxww | grep rbd` on the server running that instance while running the `lxc query` to see if LXD is running those queries?

**Comment by tomponline** [comment]
@stgraber seems so: https://github.com/lxc/lxd/issues/8834#issuecomment-852434644

"i can indeed see the rbd du process when trying to do a lxc query on snapshots."

**Comment by ninjuan11** [comment]
yes, the queries are spawned and will stay forever.

**Comment by stgraber** [comment]
Are you using ceph.external=true with the snap? It could be some issue where your ceph cluster needs a more recent version of the client than the one in the snap.

**Comment by ninjuan11** [comment]
no i'm using the internal, had massive problems with external, as it was unable to connect

**Comment by ninjuan11** [comment]
i just switched to external - good news: external is now working (issue was in version 3 of lxd, so maybe this has been fixed)
still - the same problem

**Comment by tomponline** [comment]
An explanation for this issue is here: https://github.com/lxc/lxd/pull/8839#discussion_r643810346
